CMakeLists.txt

```diff
@@ -17,6 +17,10 @@ if(APPLE)
     OPTION(APPLE_OUTPUT_DYLIB "Output dylib shared library" OFF)
 endif()
 
+if((CMAKE_CXX_COMPILER_ID STREQUAL "GNU") AND (UNIX) AND (NOT APPLE))
+  ADD_DEFINITIONS(-DGCC)
+endif()
```

Comment:
Maybe it's better to move this somewhere lower to other definitions?

Reply:
sorry, I didn't understand it, can you describe more details ?

Reply:
I mean, move these lines lower in the file. For example, somewhere around here
https://github.com/Microsoft/LightGBM/blob/606eeefeb8e1f217a0f91fa0f11137125942b549/CMakeLists.txt#L66-L68
or here
https://github.com/Microsoft/LightGBM/blob/606eeefeb8e1f217a0f91fa0f11137125942b549/CMakeLists.txt#L118-L119
where other `ADD_DEFINITIONS` are set.